src/hooks/use-filtered-market-coins.hook.ts

```diff
@@ -0,0 +1,63 @@
+import { useMemo, useState, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { MarketCoinsSortFieldEnum } from '../enums/market-coins-sort-field.enum';
+import { selectSortValue } from '../store/market/market-actions';
+import {
+  useFavouriteTokens,
+  useMarketTopCoinsWithoutTez,
+  useSortFieldSelector
+} from '../store/market/market-selectors';
+import { isString } from '../utils/is-string';
+import { sortMarketCoins } from '../utils/market.util';
+import { MarketCoin } from './../store/market/market.interfaces';
+
+export const useFilterdMarketCoins = () => {
+  const dispatch = useDispatch();
+  const tokens = useMarketTopCoinsWithoutTez();
+  const sortFiled = useSortFieldSelector();
+  const favouriteTokens = useFavouriteTokens();
+
+  const inputTypeIndexDefault = favouriteTokens.length === 0 ? 0 : 1;
+
+  const [searchValue, setSearchValue] = useState<string>();
+  const [inputTypeIndex, setInputTypeIndex] = useState<number>(inputTypeIndexDefault);
+
+  const filteredAssetsList = useMemo<Array<MarketCoin>>(() => {
+    const source = inputTypeIndex === 0 ? tokens : tokens.filter(item => favouriteTokens.includes(item.id));
+    const sortedMarketCoins = sortMarketCoins(source, sortFiled);
+
+    if (isString(searchValue)) {
+      const lowerCaseSearchValue = searchValue.toLowerCase();
+      const result: Array<MarketCoin> = [];
+
+      for (const asset of sortedMarketCoins) {
+        const { name, symbol } = asset;
+
+        if (name.toLowerCase().includes(lowerCaseSearchValue) || symbol.toLowerCase().includes(lowerCaseSearchValue)) {
+          result.push(asset);
+        }
+      }
+
+      return result;
+    } else {
+      return sortedMarketCoins;
+    }
+  }, [searchValue, sortFiled, inputTypeIndex, favouriteTokens]);
+
+  const handleSetSortField = useCallback(
+    (sortValue: string) => dispatch(selectSortValue(sortValue as MarketCoinsSortFieldEnum)),
+    []
+  );
+
+  const handleSelectorChange = (index: number) => setInputTypeIndex(index);
```

Comment:
the same with this handler https://github.com/madfish-solutions/templewallet-mobile/pull/711/files#r1052107787